src/python/pants/backend/jvm/targets/jvm_binary.py

```diff
@@ -209,7 +211,7 @@ def __init__(self, rules=None, default_dup_action=Duplicate.SKIP):
     self.payload.add_fields({
       'default_dup_action' : PrimitiveField(Duplicate.validate_action(default_dup_action))
     })
-    self._rules = assert_list(rules, expected_type=JarRule)
+    self._rules = assert_list(rules, expected_type=JarRule, field_name="_rules")
```

Comment:
The field name here is `rules`: https://pantsbuild.github.io/build_dictionary.html#jar_rules